godot-core/src/init/mod.rs

```diff
@@ -209,6 +219,20 @@ pub unsafe trait ExtensionLibrary {
     fn on_level_deinit(level: InitLevel) {
         // Nothing by default.
     }
+
+    /// Whether to enable hot reloading of this library. Return `None` to use the default behavior.
+    ///
+    /// Enabling this will ensure that the library can be hot reloaded. If this is disabled then hot reloading may still work, but there is no
+    /// guarantee. Enabling this may also lead to memory leaks, so it should not be enabled for builds that are intended to be final builds.
+    ///
+    /// By default this is enabled for debug builds and disabled for release builds.
+    ///
+    /// Note that this is only checked *once* upon initializing the library. Changing this from `true` to `false` will be picked up as the
+    /// library is then fully reloaded upon hot-reloading, however changing it from `false` to `true` is almost certainly not going to work
+    /// unless hot-reloading is already working regardless of this setting.
+    fn override_hot_reload() -> Option<bool> {
+        None
+    }
```

Comment:
I might still change this later to return `bool` and do the release/debug check in the default impl, so that the user-facing method is simpler (there is no reason to override it and return `None`).